### DISCUSSION TITLE
UIX Inconsistency

### DISCUSSION THREAD

**Discussion by MartinHugh** [EdgeTX/edgetx#2137] [General]
I am running v2.5.0 and see an inconsistency in the UI.
I have checked the diff between 2.7.1 and 2.5.0 and cannot see that this has been addressed.

I am not in a position to install the latest version to confirm this inconsistency, and rather than raise a bug report immediately on 2.5.0 I wanted to check that this has not been addressed.

If I want to select a model, from the Select Model page, the model to be selected is delineated in a solid frame.  The other models are not delineated so it is obvious which one I am selecting.

If I go to edit screen widgets, I need to select one to bring up the submenu to select "Widget Settings".
However, all the widgets which are **not** selected are **each delineated in a solid frame**.  The one to be selected is in a faint dotted white frame.

I cant tell you how often when I am at the flying field trying to adjust a widget I get this wrong.  This is particularly likely when there are only two widgets set and one is framed and one isnt.

Is this still the way 2.7.1 works.  
Is there a rationale for this inconsistency
If Yes and No, I will be happy to raise a bug report

Thanks for reading.

**Comment by rdeanchurch** [comment]
Yes it works the same in V 2.7.1.
I agree it is not consistent and would be better if there were the same indications in each case.

**Comment by pfeerick** [comment]
As @rdeanchurch said, it's the same in v2.7.1 as there has been little change to the UI in that area. In 2.8 due to the LVGL re-write it has changed slightly, but also has a lot the same... the edit screen is basically the same. When at the main screen, iyou have to press and hold ENT if using the roller to interact with widgets as opposed to the main screen. I do agree, it would be better to have a solid frame, and dotted frames for the inactive (or perhaps use active and inactive frame colours) - and would have sworn I opened an issue about that... but can't see it now. So by all means open an issue and reference this discussion (#2137) and suggest the widget edit screen be improved. It could also be used on the mains widget screen as there is no "reason" it needs to be a dotted focus highlight as it could be solid and flash to shown it's going to loose focus just the same. 

btw, you can see all of this if you download a test version of Companion [from github](https://github.com/EdgeTX/edgetx/actions) and use the simulator, so you don't have to worry about messing up your radio and settings ;) And if you sync your SD card, you can even see it with your transmitter colour theme and widget settings.

**Reply by MartinHugh** [reply]
> As @rdeanchurch said, it's the same in v2.7.1 as there has been little change to the UI in that area. In 2.8 due to the LVGL re-write it has changed slightly, but also has a lot the same... the edit screen is basically the same. When at the main screen, iyou have to press and hold ENT if using the roller to interact with widgets as opposed to the main screen. I do agree, it would be better to have a solid frame, and dotted frames for the inactive (or perhaps use active and inactive frame colours) - and would have sworn I opened an issue about that... but can't see it now. So by all means open an issue and reference this discussion (#2137) and suggest the widget edit screen be improved. It could also be used on the mains widget screen as there is no "reason" it needs to be a dotted focus highlight as it could be solid and flash to shown it's going to loose focus just the same.
> 
> btw, you can see all of this if you download a test version of Companion [from github](https://github.com/EdgeTX/edgetx/actions) and use the simulator, so you don't have to worry about messing up your radio and settings ;) And if you sync your SD card, you can even see it with your transmitter colour theme and widget settings.

Thanks @pfeerick and @rdeanchurch. I will raise an issue for this.

**Comment by rdeanchurch** [comment]
> btw, you can see all of this if you download a test version of Companion from github

Easy for you to say.  I can't even find Companion or EdgeCompanon on github...let alone
a 'download' button.  My search efforts are not very good obviously.

**Reply by pfeerick** [reply]
For release versions of Companion, it's at the bottom of the release notes for the release, along with the firmware. For development builds, you would click on the "Actions" tab at the top (which I linked directly to before), pick the action for the OS flavor of Companion you want (Linux, Mac, Windows)... and then look for a recent release (that was successfully built) on the `main` branch. And the download link / artifact is then at the bottom.

i.e. Like this. I right clicked and chose "save link as" so I have control over where and what filename used, but you can also click normally and it'll save to your default download folder. 

https://user-images.githubusercontent.com/5500713/179437463-5f5f233f-0b7f-41af-9d95-a0455f842e47.mp4

**Comment by rdeanchurch** [comment]
Thank you for the help and your patience.
I finally got it.

<http://www.rcgroups.com/forums/member.php?u=139391>
 <http://www.rcgroups.com/forums/member.php?u=139391> <http://www.rcgroups.com/forums/member.php?u=139391>
<http://openrcforums.com/forum/viewtopic.php?f=25&t=185&sid=86567a7056f8c66cb2a565b2f73df302#wrap>

________________________________
